airbyte-integrations/connectors/source-freshdesk/source_freshdesk/source.py

```diff
@@ -34,13 +34,14 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
         try:
             url = urljoin(f"https://{config['domain'].rstrip('/')}", "/api/v2/settings/helpdesk")
             response = requests.get(url=url, auth=self._create_authenticator(config["api_key"]))
-            if not response.ok:
-                alive = False
-                try:
-                    body = response.json()
-                    error_msg = f"{body.get('code')}: {body['message']}"
-                except ValueError:
-                    error_msg = "Invalid credentials"
+            response.raise_for_status()
+        except requests.HTTPError as error:
+            alive = False
+            try:
+                body = error.response.json()
+                error_msg = f"{body.get('code')}: {body['message']}"
+            except ValueError:
+                error_msg = "Invalid credentials"
```

Comment:
In which situation `ValueError` is raised here?
If whatever non HTTPError is raised is I don't think the cause would be `Invalid Credential`.
I think this try except block is redundant as non HttpError are watched by the higher level `except Exception as error:`

Reply:
You're right. I've removed the handling of this kind of error.